warp10/src/main/java/io/warp10/script/functions/TOPICKLE.java

```diff
@@ -16,29 +16,45 @@
 
 package io.warp10.script.functions;
 
+import io.warp10.continuum.gts.GTSEncoder;
+import io.warp10.continuum.gts.GTSPickler;
+import io.warp10.continuum.gts.GeoTimeSerie;
 import io.warp10.script.NamedWarpScriptFunction;
 import io.warp10.script.WarpScriptException;
 import io.warp10.script.WarpScriptStack;
 import io.warp10.script.WarpScriptStackFunction;
 
 import java.io.IOException;
 
+import net.razorvine.pickle.Pickler;
 import net.razorvine.pyro.serializer.PickleSerializer;
 
 /**
  * Encode an object as Pickle
  */
 public class TOPICKLE extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+  private static final TOGTS TOGTS = new TOGTS("TOGTS");
   
   public TOPICKLE(String name) {
     super(name);
   }
   
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
-    Object o = stack.pop();
+    Object o = stack.peek();
+
+    if (o instanceof GTSEncoder) {
+      stack = (WarpScriptStack) TOGTS.apply(stack);
+    }
+
+    o = stack.pop();
     
     try {
+      // register pickler for custom classes
+      Pickler.registerCustomPickler(GeoTimeSerie.class, new GTSPickler());
```

Comment:
Can this be factored out? Is GTSPickler MT safe?

Reply:
I don't think it can be factored out if you are using the same serializer for other objects that may contain geo time series or be contained by geo time series.

Reply:
The Serializer is thread safe IIRC

Reply:
No it doesn't seem thread safe: [Pickler](https://github.com/irmen/Pyrolite/blob/master/java/src/main/java/net/razorvine/pickle/Pickler.java) used by [PickleSerializer](https://github.com/irmen/Pyrolite/blob/master/java/src/main/java/net/razorvine/pyro/serializer/PickleSerializer.java) is not

Reply:
The call ro registerCustomPickler at each call to apply does not seem right, the registration is probably to be done once and for all since your GTSPickler is MT safe

Reply:
Sure. I didn't get that was what you meant by "factored out".